company/validation.go

```diff
@@ -1,17 +1,48 @@
 package company
 
 import (
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-
-	pb "github.com/dinhtp/lets-go-pbtype/company"
+    "strings"
 
+    pb "github.com/dinhtp/lets-go-pbtype/company"
+    "google.golang.org/grpc/codes"
+    "google.golang.org/grpc/status"
 )
 
 func validateOne(r *pb.OneCompanyRequest) error {
-	if "" == r.GetId() {
-		return status.Error(codes.InvalidArgument, "company id is required")
-	}
+    if "" == r.GetId() {
+        return status.Error(codes.InvalidArgument, "company id is required")
+    }
+    return nil
+}
+
+func validateCreate(r *pb.Company) error {
+    if r.GetName() == "" || r.GetPhone() == "" || r.GetEmail() == "" || r.GetAddress() == "" || r.GetTaxNumber() == "" {
+        return status.Error(codes.InvalidArgument, "Attributed is required")
+    }
+    return nil
+}
+
+func validateUpdate(r *pb.Company) error {
+    if "" == r.GetId() {
+        return status.Error(codes.InvalidArgument, "company id is required")
+    }
+    return validateCreate(r)
+}
+
+func validateList(r *pb.ListCompanyRequest) error {
+    var page = r.GetPage()
+    var limit = r.GetLimit()
+    var field = strings.Split(r.GetSearchField(), ",")
+
+    if (page <= 0) || (limit <= 0) {
+        return status.Error(codes.InvalidArgument, "Invalid Attribute")
+    }
 
-	return nil
+    for i := 0; i < len(field); i++ {
```

Comment:
Đoạn này trước khi vào for loop thì check xem có search field và search value ko, nếu không có thì return nil luôn. ko cần validate nữa
có giá trị thì mình mới validate